packages/eslint-config/shared.js

```diff
@@ -68,33 +75,52 @@ module.exports = {
       'error',
       {
         'newlines-between': 'always',
-        pathGroupsExcludedImportTypes: ['react'],
+        // We set this to an empty array to override the default value, which is `['builtin', 'external', 'object']`.
+        // Judging by the number of issues on the repo, this option seems to be notoriously tricky to understand.
+        // From what I can tell, if the value of this is `['builtin']` that means it won't sort builtins.
+        // But we have a rule for builtins below (react), so that's not what we want.
+        //
+        // See...
+        // - https://github.com/import-js/eslint-plugin-import/pull/1570
+        // - https://github.com/import-js/eslint-plugin-import/issues/1565
+        pathGroupsExcludedImportTypes: [],
+        // Only doing this to add internal. The order here maters.
+        // See https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md#groups-array
+        groups: [
+          'builtin', // 0
+          'external', // 1
+          'internal', // 2
+          'parent', // 3
+          'sibling', // 4
+          'index', // 5
+        ],
         pathGroups: [
           {
             pattern: 'react',
             group: 'builtin',
-            position: 'after',
+            position: 'after', // 0.1 -> .9
           },
           {
             pattern: '@redwoodjs/**',
             group: 'external',
-            position: 'after',
-          },
-          {
-            pattern: 'src/lib/test',
-            group: 'parent',
-            position: 'before',
+            position: 'after', // 1.1 -> 1.9
           },
           {
-            pattern: 'src/**',
-            group: 'parent',
-            position: 'before',
+            // Matches src/directives/**/*.{js,ts}, src/services/**/*.{js,ts}, src/graphql/**/*.sdl.{js,ts}
+            pattern: 'src/*/\\*\\*/\\*.*\\{js,ts\\}',
```

Comment:
```suggestion
            pattern: 'src/*/\\*\\*/\\*\\.[^/]*\\{js,ts\\}$',
```

I escaped the dot to force the pattern to literally be `/**/*.`
After that dot we allow anything that isn't a slash 0 or more times. And then it has to end with `{js,ts}`

This should be fine on Windows too, because even on Windows you specify import paths with a single forward slash.